src/main/java/com/limechain/storage/trie/TrieStorage.java

```diff
@@ -515,4 +515,55 @@ private void insertTrieNodeStorage(InsertTrieNode trieNode,
 
         db.save(key, storageValue);
     }
+
+    /**
+     * Loads the trie structure from the database starting from a given root.
+     *
+     * @param trieRoot The Merkle root of the trie to load.
+     * @return The reconstructed trie structure, or null if the root is not found.
+     */
+    public TrieStructure<NodeData> loadTrieStructure(byte[] trieRoot) {
+        if (trieRoot == null || trieRoot.length == 0) {
+            log.warning("Invalid or empty trie root provided.");
+            return null;
+        }
+
+        TrieNodeData rootNode = getTrieNodeFromMerkleValue(trieRoot);
+        if (rootNode == null) {
+            log.warning("No trie node found for the given root.");
+            return null;
+        }
+
+        TrieStructure<NodeData> trie = new TrieStructure<>();
+        loadSubTrie(trie, rootNode, trieRoot, Nibbles.EMPTY);
+        return trie;
+    }
+
+    /**
+     * Recursively loads the sub-trie from the database and adds it to the given trie structure.
+     *
+     * @param trie            The trie structure being constructed.
+     * @param currentNodeData The current node data being processed.
+     * @param merkleValue     The Merkle value of the current node.
+     * @param currentPath     The nibble path to the current node.
+     */
+    private void loadSubTrie(TrieStructure<NodeData> trie, TrieNodeData currentNodeData, byte[] merkleValue,
+                             Nibbles currentPath) {
+        trie.insertNode(currentPath,
```

Comment:
Known issue: we're inserting *all nodes* from the DB as key-value-pairs, thus marking a lot of structural branch nodes (i.e. branch nodes with no storage value) as actually having a storage value.

Miraculously though, we're still able to calculate proper state root hashes since we're not actually utilising our knowledge of which node has a storage value anywhere in our algorithm, which is a grand coincidence.

P.S. On another note, inserting `currentNodeData.getTrieRootRef()` instead will always result in weird behaviours where an actual branch node coming from the DB would be inserted as a storage node with a NodeData(value = null; merkleValue = {correctMerkleValue})... a weird form of invalid state, fortunately not a blocker for block execution with our current implementation.